lib/network.js

```diff
@@ -42,16 +42,33 @@ const fetchPolyfill = (url, options = {}) => {
   });
 };
 
-const ipToInt = (ip = '127.0.0.1') => {
-  if (ip === '') return 0;
+const IPV4_OCTETS = 4;
+
+const ipToInt = (ip) => {
+  if (typeof ip !== 'string') return Number.NaN;
   const bytes = ip.split('.');
+  if (bytes.length !== IPV4_OCTETS) return Number.NaN;
   let res = 0;
   for (const byte of bytes) {
-    res = (res << 8) + parseInt(byte, 10);
+    res = res * 256 + parseInt(byte, 10);
```

Comment:
What do you think about leaving the shifts and adding 2^32 to res after the cycle 'for' instead of multiplying?

Reply:
The most handy format is * 256, see test cases

Reply:
My mistake, it's not just res + 2^32, it's something like this:
res = (res + Math.pow(2, 32) & MAX_32_BIT;

Reply:
This means that for this to work we need to make the real sum modulo 'pow(2, 32)'
```js
res = (POW2_32 + res) % POW2_32;
```
or use the ternary operator
```js
res = res < 0 ? (POW2_32 + res) : res ;
```
but it's probably not nice.